### DISCUSSION TITLE
Android: Plans to support 16kb page sizes

### DISCUSSION THREAD

**Discussion by phillippschmedt** [dotnet/maui#23331] [General] [answered]
Android 15 will support 16 KB page sizes, see: https://developer.android.com/guide/practices/page-sizes

Running an example MAUI app on an Android 15 emulator with 16KB page sizes will crash the app on startup. It is to be expected that we will soon see devices that will run Android 15 and 16KB page sizes.

What is the current plan on 16KB page sizes support?

**Comment by Felix-Dev** [comment, marked as answer]
Looks like the .NET for Android team is already on this:

1. https://github.com/dotnet/android/pull/9046
2. https://github.com/dotnet/android/wiki/Android-support-for-devices-with-16k-pages
3. https://github.com/dotnet/runtime/issues/103360

**Comment by phillippschmedt** [comment]
There still seem to be problems with Android 16kb. https://github.com/dotnet/maui/issues/27995

**Reply by danieltharris** [reply]
Is this now working? Supporting 16kb page sizes is required from May 2026 for all apps targeting Android 15+

**Reply by giuseppenovielli** [reply]
HI,
yes it work -> https://github.com/dotnet/android/issues/9829#issuecomment-3145108321
Test with >= .NET Android 35.0.78

**Reply by MitchBomcanhao** [reply]
> Is this now working? Supporting 16kb page sizes is required from May 2026 for all apps targeting Android 15+

@danieltharris  may 2026? I think you mean November 1st, 2025...

**Reply by danieltharris** [reply]
> > Is this now working? Supporting 16kb page sizes is required from May 2026 for all apps targeting Android 15+
> 
> 
> 
> @danieltharris  may 2026? I think you mean November 1st, 2025...

Yes i think you're right. May 1 2026 is the cut off for updating an app that doesn't currently support 16kb page sizes, from Google email:

`To ensure your app works correctly on the latest versions of Android, Google Play requires all apps targeting Android 15+ to support 16 KB memory page sizes.

From May 1, 2026, if your app updates do not support 16 KB memory page sizes, you won't be able to release these updates.`

**Reply by MitchBomcanhao** [reply]
`Starting November 1st, 2025, all new apps and updates to existing apps submitted to Google Play and targeting Android 15+ devices must support 16 KB page sizes on 64-bit devices` https://developer.android.com/guide/practices/page-sizes

confusing messaging from google. I'd take the developer page over some random email.

**Reply by beeradmoore** [reply]
You can also request an extension to May 1st 2026.

**Reply by gabsamples6** [reply]
Is the apk analyzer enough to detect if you have non compatible native library- it looks like its another way to quickly assess it

**Reply by Glopex** [reply]
hoping for an implementation soon 🙏
